### PR TITLE
Provide extra submission metadata for judges

### LINKED ISSUE #802: Provide extra submission metadata for judges
Notably, allowing judges access to a hash of the submitter's username as well as attempt number would be light in terms of computation and protocol changes, but allow for some more interesting grading schemes (e.g. running on different test data for different attempts).

### PR DESCRIPTION
**Pull request by Ninjaclasher**
Fixes #802.

### CODE REVIEW

File: judge/bridge/judgecallback.py [hunk 1]
```diff
@@ -70,12 +71,15 @@ def get_related_submission_data(self, submission):
             ))
             return
 
+        attempt_no = Submission.objects.filter(problem__id=pid, contest__participation__id=part_id,
```
Comment (Xyene) [inline judge/bridge/judgecallback.py]: Pretty sure this shouldn't include CE/IE.

File: judge/bridge/judgecallback.py [hunk 2]
```diff
@@ -70,12 +71,15 @@ def get_related_submission_data(self, submission):
...
+                                               user__id=uid, date__lt=sub_date).count() + 1
+
         try:
             time, memory = (LanguageLimit.objects.filter(problem__id=pid, language__id=lid)
                                          .values_list('time_limit', 'memory_limit').get())
         except LanguageLimit.DoesNotExist:
             pass
-        return time, memory, short_circuit, is_pretested
+        return time, memory, short_circuit, is_pretested, part_virtual, attempt_no, uid
```
Comment (Xyene) [inline judge/bridge/judgecallback.py]: Can we define a `namedtuple` for this?

File: judge/bridge/judgehandler.py [hunk 3]
```diff
@@ -104,7 +104,7 @@ def working(self):
         return bool(self._working)
 
     def get_related_submission_data(self, submission):
-        return 2, 16384, False, False
+        return 2, 16384, False, False, None, 1, None
```
Comment (Xyene) [inline judge/bridge/judgehandler.py]: This tuple is really getting out of hand...

**Comment by Ninjaclasher** [comment]
Turns out I was accidentally filtering _for_ CEs and IEs, instead of excluding them. This should be fixed in the latest commit. Sorry.

**Pull request merged**